dashboard/app/controllers/aichat_controller.rb

```diff
@@ -15,24 +15,46 @@ def chat_completion
     end
 
     # Check for PII / Profanity
-    # Copied from ai_tutor_interactions_controller.rb - not sure if filtering is working.
     locale = params[:locale] || "en"
-    new_message_text = params[:newMessage]
-    # Check only the newest message from the user for inappropriate content.
-    filter_result = ShareFiltering.find_failure(new_message_text, locale) if new_message_text
-    # If the content is inappropriate, we skip sending to endpoint and instead hardcode a warning response on the front-end.
-    return render(status: :ok, json: {status: filter_result.type, flagged_content: filter_result.content}) if filter_result
-
-    input_json = AichatSagemakerHelper.format_inputs_for_sagemaker_request(params[:aichatModelCustomizations], params[:storedMessages], params[:newMessage])
-    sagemaker_response = AichatSagemakerHelper.request_sagemaker_chat_completion(input_json)
+    filter_result = ShareFiltering.find_failure(params[:newMessage], locale)
+    if filter_result&.type == 'profanity'
+      new_messages = [
+        {
+          role: 'user',
+          content: params[:newMessage],
+          status: SharedConstants::AI_INTERACTION_STATUS[:PROFANITY_VIOLATION]
+        }
+      ]
+      session_id = log_chat_session(new_messages)
+      return render(
+        status: :ok,
+        json: {
+          status: filter_result.type,
```

Comment:
Should we convert this into an `AI_INTERACTION_STATUS`?

Reply:
I dunno...it's a different string ("profanity" vs "profanity_violation"), and feels slightly different (ie, not an interaction status, but a violation type that will also potentially include things like "address", etc that won't match 1:1 with an interaction status)

Reply:
Gotcha, yeah maybe a follow-up then but it would be good to have some sort of constant mapping for this on the front end too (rather than just checking the string `'profanity'`?)